lib/deploy/events/cloudWatchEvent/compileCloudWatchEventEvents.js

```diff
@@ -92,12 +94,12 @@ module.exports = {
                     ${InputPath ? `"InputPath": "${InputPath.replace(/\r?\n/g, '')}",` : ''}
                     "Arn": { "Ref": "${stateMachineLogicalId}" },
                     "Id": "${cloudWatchId}",
-                    "RoleArn": {
+                    ${IamRole ? `"RoleArn":"${IamRole}"` : `"RoleArn": {
```

Comment:
this feels a bit weird, having the entire blob inside an if/else, can you do the if-else on ln47 when you assign the value of IamRole instead?

Reply:
@theburningmonk  I'll do my best)